docs/openapi-docs/src/test/resources/expected_nullable_option_class_field.yml

```diff
@@ -0,0 +1,54 @@
+openapi: 3.1.0
+info:
+  title: ClassWithOptionClassField
+  version: '1.0'
+paths:
+  /:
+    post:
+      operationId: postRoot
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ClassWithOptionClassField'
+        required: true
+      responses:
+        '200':
+          description: ''
+          content:
+            text/plain:
+              schema:
+                type: string
+        '400':
+          description: 'Invalid value for: body'
+          content:
+            text/plain:
+              schema:
+                type: string
+components:
+  schemas:
+    Bar:
+      required:
+        - bar
+      type: object
+      properties:
+        bar:
+          type: integer
+          format: int32
+    ClassWithOptionClassField:
+      required:
+        - requiredStringField
+      type: object
+      properties:
+        optionalObjField:
+          required:
```

Comment:
it'll require changes in the `sttp-apispec` for all encoders.
I could do it for the `circe` integration only. will it be acceptable?

Reply:
- PR for `sttp-apispec` circe is ready https://github.com/softwaremill/sttp-apispec/pull/118
- updated current PR to generete `oneOf` instead of inlining